### ISSUE TITLE
Getting the pure JS version working in IE11

### ISSUE THREAD

**Issue by SteveEdson** [uber/deck.gl#2145]
I'm using the `6.0.2` version of the `deckgl.min.js` file, imported with:

```js
import deck from '../lib/deck.min';
```

However, the code contains ES6 language features and isn't compatible with IE11. Can anyone tell me how I can create a new version of this code, compiled to ES5 please? I've been trying to transpile the minified file, but that doesn't work, I just get errors saying `n is not a function`.

Alternatively, I don't specifically need to be using the `6.x` releases, is there an older version of deck.gl that is compatible with IE 11 (also tried v5 with no luck).

Thanks

**Comment by ibgreen** [comment]
I would expected that the script version of deck.gl would be completely transpiled to ES5.

However, even if all syntax is transpiled, I don't think we include any polyfills for missing features in built-in classes. For non-evergreen browsers like IE11, such polyfills are needed and should be included before the deck.gl script.

Ideally we should find a script tag that publishes babel 7 polyfills. I only searched quickly found this so far https://cdnjs.com/libraries/babel-polyfill.

FWIW, v5.3 would likely have a better shot at working on IE11 than v6.

Give it a shot and let us know.

**Comment by darionco** [comment]
There are a few things that are not transpiled by buble, I can explain how to make it compile and run, but to my experience there is a rendering bug in IE11. You need to compile from source. Checkout deck.gl and then:

- Open terminal and navigate to the `examples/get-started/pure-js` folder within the repo
- Add `whatwg-fetch` to your dependencies
- Add the following modules to devDependencies 
  - babel-core
  - babel-loader
  - babel-plugin-transform-object-rest-spread
  - babel-polyfill
  - babel-preset-latest
  - babel-preset-react
  - brfs
  - transform-loader
- In `webpack.config.js`:
  - Change the `entry` to `['babel-polyfill', 'whatwg-fetch', resolve('./app.js')]`
  - Add a `devtool` if you need debugging: `devtool: 'inline-source-map',`
  - Uncomment the last line: `module.exports = env => (...`
  - Add module building rules: 
```
    module: {
        rules: [
            {
                test: /bindings\/bindings\.js/,
                loader: 'transform-loader?brfs',
            },
            {
                // Compile ES2015 using buble
                test: /\.js$/,
                loader: 'babel-loader',
                include: [resolve('.')],
                exclude: [/node_modules/],
                options: {
                    presets: [
                        [ 'latest', { es2015: { modules: false } } ],
                    ],
                    plugins: [
                        'transform-object-rest-spread',
                    ],
                }
            }
        ]
    },
```
- You can no build from the command line with: `yarn start-local` or `npm run start-local`

**Comment by ibgreen** [comment]
Thanks. Some info:

* The script file is built in `modules/lite`. The webpack config and babelrc are there.
* While the examples transpile using buble, the lib files are transpiled using babel.
* By convention, we don't use object-rest-spread syntax in the deck.gl code base (it is not part of ES6). Did you actually need that transform, or you just copied an existing set?

**Comment by darionco** [comment]
Yes it is needed specifically for the example, in `mapbox.js` with the example folder,  line `12-13`:
```
    this._map = new mapboxgl.Map({
      ...props,
```

**Comment by darionco** [comment]
The need for `transform-object-rest-spread` seems to have been dropped in 79ea9201b525313b412ad94b2f317ecb02ec7e34

**Comment by darionco** [comment]
What version of deck.gl are you using?
I am using the ArcLayer and it works fine in `5.3.1` but not in >= `6.0.0`

**Comment by ibgreen** [comment]
@SteveEdson This looks pretty good. 

I have seen the missing top caps in other examples, in those cases they were related to a luma.gl issue in v6.0. 

We have started to modernize the way we use WebGL to improve performance and it has caused some regressions on older browsers.  I have an in-progress fix here: https://github.com/uber/luma.gl/pull/618, trying to find the time to wrap it up.

The tearing effect is due to Z fighting, a common 3D / depth buffer artifact. There are case-by-case techniques for dealing with it mentioned e.g. in the deck.gl tips and tricks page, but in this case I'd probably ignore it since when the top caps are back you shouldn't see these interior walls anyway.

**Comment by SteveEdson** [comment]
Thanks for the update. Switching back to 5.3 has helped with a lot of the rending issues in IE11.

Just a couple of questions though, previously when using v6, I initalised Deck.gl with this in the options:

```js
controller: {
    scrollZoom: false,
    touchRotate: true,
},
``` 

this didn't work using 5.3, so I had to switch to use:

```js
controller: MapController,
```

This works, but now I'm not sure how to set the `scrollZoom` and `touchRotate` values?

Thanks

**Comment by SteveEdson** [comment]
Just figured this out. Had a look at the code and doing this fixes it:

```js
scrollZoom: false,
controller: MapController,
```

It looks like it just merges in the main props.

Thanks.
